scripts/Create-StorageContainer.ps1

```diff
@@ -71,59 +75,91 @@ trap [Exception]
     # this should trigger and catch and resume
     throw "Script execution failed: $($_)";
 }
+function New-Containers
+{
+    param(
+            [Parameter(Mandatory=$true)][string]$ContainerNames
+         )
+    # Create the container
+    [array]$storageContainerList = $ContainerNames.Split(",");
+    if ($storageContainerList.Length -gt 0)
+    {
+        Log-Message "Creating Multiple Storage Containers (Cli: $AzureCliVersion): $($ContainerNames)" -Context "Create Storage Containers"
+    }
 
-#########################
-#
-# ENTRY POINT
-#
-#########################
+    foreach($storageContainerName in $storageContainerList)
+    {
+        # todo: add retries for better resiliency
+        Log-Message "Creating Storage Container (Cli: $AzureCliVersion): $($storageContainerName)" -Context "Create Storage Containers"
 
-$invocation = (Get-Variable MyInvocation).Value 
-$currentPath = Split-Path $invocation.MyCommand.Path 
-Import-Module "$($currentPath)/Common.ps1" -Force
+        # Check if container already exists
+        $status = Get-ContainerExists -Container $storageContainerName
 
-# Login First & set context
-Authenticate-AzureRmUser -AadWebClientId $AadWebClientId -AadWebClientAppKey $AadWebClientAppKey -AadTenantId $AadTenantId;
-Set-AzureSubscriptionContext -AzureSubscriptionId $AzureSubscriptionId
+        if (!$status)
+        {
+            Log-Message "creating '$storageContainerName' container"
 
-# Create the container
-[array]$storageContainerList = $StorageContainerNames.Split(",");
+            # todo: fall back to azure cli since there are existing issues with installation of azure powershell cmdlets for linux
```

Comment:
- [ ] nit: i think the TODO has been satisfied. If you'd like, you can remove it now.